src/Workleap.DomainEventPropagation.Subscription.PullDelivery/EventPullerService.cs

```diff
@@ -63,6 +66,15 @@ private class EventGridSubscriptionEventPuller
             this._serviceScopeFactory = serviceScopeFactory;
             this._logger = logger;
             this._eventGridTopicSubscription = eventGridTopicSubscription;
+
+            if (eventGridTopicSubscription.RetryDelays != null)
+            {
+                this._retryDelays = eventGridTopicSubscription.RetryDelays
```

Comment:
This looks more complicated than the previous commit (using an array). Is there a reason to convert a list to a dictionary?

Reply:
It's an idea I got while waiting for the review. It's basically precompiling the mapping between the deliveryCount and the delays so the code later on doesn't have to play around with indexes and limits.